### PR TITLE
Feature/mc 9573 Administration UI for OpenID Connect providers

### PR DESCRIPTION
**Pull request by pjmonks**
* Feature toggle and API property added to control visibility of OpenID Connect provider list in administrator pages
* Table view of available providers with add/edit/delete options
* Add/edit form for management of providers

Known issues:

* Cannot update a standard provider that already exists - see MauroDataMapper-Plugins/mdm-plugin-authentication-openid-connect#1

### CODE REVIEW

**Comment by olliefreeman** [comment]
The plugin issue noted as a blocker has been resolved … I included a test to ensure it doesn’t happen again

File: src/app/admin/openid-connect-provider/openid-connect-provider.component.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+    const payload = this.form.createPayload();
+    const request: Observable<OpenIdConnectProvidersDetailResponse> = this.editExisting
+      ? this.resources.pluginOpenIdConnect.update(this.id, payload)
+      : this.resources.pluginOpenIdConnect.save(payload);
+
+    request
+      .pipe(
+        catchError(error => {
+          this.messageHandler.showError('There was a problem saving the OpenID Connect provider.', error);
```
Comment (gammonpeter) [inline src/app/admin/openid-connect-provider/openid-connect-provider.component.ts:96]: I added some junk info and tried to save got the error message but also directed to a 500 screen, we might need to think about how we handle these

**Review by gammonpeter — CHANGES_REQUESTED** [review]

File: src/app/admin/openid-connect-provider-table/openid-connect-provider-table.component.html [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        fxFlex
+        fxLayout="row"
+        fxFlex="20"
+        fxFlex.md="20"
+        fxFlex.sm="25"
+        fxFlex.xs="100"
+        fxLayoutAlign="flex-end center"
+        fxLayoutAlign.xs="flex-start center"
+      >
+        <mdm-data-type-list-buttons
```
[the remark this answers is not in the document]
Comment (pjmonks) [reply src/app/admin/openid-connect-provider-table/openid-connect-provider-table.component.html:55]: You're right, corrected this

File: src/app/admin/openid-connect-provider/openid-connect-provider.component.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+          return EMPTY;
+        })
+      )
+      .subscribe(() => {
+        this.messageHandler.showSuccess('Property was updated successfully.');
```
[the remark this answers is not in the document]
Comment (pjmonks) [reply src/app/admin/openid-connect-provider/openid-connect-provider.component.ts:101]: Sorry, a copy-paste mistake. Fixed.

File: src/style/_default.scss [hunk 4]
```diff
@@ -280,4 +280,10 @@ ngx-skeleton-loader .loader {
 
 .mat-form-field-appearance-outline .mat-form-field-infix {
   padding: .6em .2em 1em .2em!important;
+}
+
+.mdm-openid-connect-icon {
```
[the remark this answers is not in the document]
Comment (pjmonks) [reply src/style/_default.scss:285]: image

Seems OK, for now.

File: src/app/admin/openid-connect-provider/openid-connect-provider.component.ts (hunk 1 above)
Comment (pjmonks) [reply src/app/admin/openid-connect-provider/openid-connect-provider.component.ts:96]: Sounds like a wider issue. This error handling is consistent with other pages, so I think I'll leave this for now and assume backend won't throw these kinds of errors.

**Review by gammonpeter — APPROVED** [review]
Looks good

**Pull request merged**